tests/onnxruntime/nightly_test_trainer.py

```diff
@@ -0,0 +1,1118 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+import tempfile
+import unittest
+from typing import List, Optional
+from unittest.mock import Mock, patch
+
+import nltk
+import numpy as np
+from datasets import load_dataset, load_metric
+from transformers import (
+    AutoConfig,
+    AutoModelForSeq2SeqLM,
+    AutoModelForSequenceClassification,
+    AutoModelForTokenClassification,
+    AutoTokenizer,
+    DataCollatorForSeq2Seq,
+    DataCollatorForTokenClassification,
+    DataCollatorWithPadding,
+    IntervalStrategy,
+    Seq2SeqTrainingArguments,
+    TrainingArguments,
+    default_data_collator,
+    is_torch_available,
+)
+from transformers.testing_utils import (
+    require_deepspeed,
+    require_optuna,
+    require_ray,
+    require_sigopt,
+    require_torch,
+    require_torch_bf16,
+    require_torch_gpu,
+    require_wandb,
+    slow,
+)
+from transformers.trainer_utils import (
+    default_hp_space_optuna,
+    default_hp_space_ray,
+    default_hp_space_sigopt,
+    default_hp_space_wandb,
+)
+from transformers.training_args import OptimizerNames
+from transformers.utils import is_apex_available, is_bitsandbytes_available
+
+
+if is_torch_available():
+    import torch
+    import transformers.optimization
+
+from optimum.onnxruntime import ORTSeq2SeqTrainer, ORTTrainer
+from optimum.utils.testing_utils import require_hf_token, require_ort_training, require_sigopt_token_and_project
+from parameterized import parameterized
+
+
+nltk.download("punkt")
+
+_MODELS_TO_TEST = {
+    ("bert", "bert-base-cased"),
+    ("distilbert", "distilbert-base-cased"),
+    ("roberta", "roberta-base"),
+    ("gpt2", "gpt2"),
+}
+
+_SEQ2SEQ_MODELS_TO_TEST = {
+    ("t5", "t5-small"),
+    ("bart", "facebook/bart-base"),
+}
+
+_TASKS_DATASETS_CONFIGS = {
+    "sequence-classification": {
+        "dataset": ["glue", "sst2"],
+        "metric": ["glue", "sst2"],
+        "data_collator": default_data_collator,
+        "data_collator_class": DataCollatorWithPadding,
+    },
+    "token-classification": {
+        "dataset": ["conll2003"],
+        "metric": ["seqeval"],
+        "data_collator_class": DataCollatorForTokenClassification,
+    },
+}
+
+_SEQ2SEQ_TASKS_DATASETS_CONFIGS = {
+    "seq2seq-lm": {
+        "dataset": ["xsum"],
+        "metric": ["rouge"],
+        "data_collator_class": DataCollatorForSeq2Seq,
+    }
+}
+
+
+def _get_models_to_test(model_list, task_list, excluded: Optional[List[str]] = None):
+    models_to_test = []
+
+    for name, model_name in model_list:
+        for feature, data_metric_config in task_list.items():
+            if excluded and name in excluded:
+                continue
+            models_to_test.append((f"{name}_{feature}", model_name, feature, data_metric_config))
+
+    return sorted(models_to_test)
```

Comment:
Why do we sort `models_to_test` ?

Reply:
I guess we want the error information to be ordered by alphabet to easier locate it or something 👀